src/common/inference/service.rs

```diff
@@ -42,15 +45,37 @@ pub struct InferenceRequest {
     pub(crate) token: Option<String>,
 }
 
-#[derive(Debug, Serialize)]
+#[derive(Debug, Serialize, Clone)]
 pub struct InferenceInput {
     data: Value,
     data_type: String,
     model: String,
     options: Option<HashMap<String, Value>>,
 }
 
+impl InferenceInput {
+    /// Tries to parse the input's options into BM25 config and returns Ok(Some(..)) if found and Ok(None) if not.
+    ///
+    /// Returns an error if bm25 has been explicitly enabled but the options could not be deserialized properly into bm25 config.
+    pub fn try_parse_bm25_config(&self) -> Result<Option<Bm25Config>, StorageError> {
+        let Some(options) = self.options.as_ref() else {
+            return Ok(None);
+        };
+
+        if options.get("use_bm25") != Some(&Value::Bool(true))
```

Comment:
Why is this here and why it has hard-coded strings?

Reply:
You're right, it should be a constant instead of directly coded into the code.

I think it's better to explicitly state that we want to use bm25, instead of "guessing" and always trying to parse the `options` into Bm25Config. Therefore we have this additional check when trying to read BM25 configuration from the inputs `options`.